app/controllers/courses_controller.rb

```diff
@@ -58,7 +58,7 @@ def statistics
       if current_user.id == @teacher_id
         @teacher = true
       elsif current_user.student?
-        @student = @terms.first.students.find_by user_id: current_user.id
+        @student = @terms.collect { |term| term.students.where(user_id: current_user.id) }.any?
```

Comment:
I've tested with 
```
Benchmark.measure { @terms.collect { |term| term.students.where(user_id: current_user.id) }.any? }
Benchmark.measure { @current_user.subscriptions.where(term_id: @course.terms.pluck(:id)).any? }
``` 
and here's what I've got: [Query benchmark.txt](https://github.com/auduchinok/hwproj/files/381381/Query.benchmark.txt)

It appears that 'collect' variant is faster on smaller data while 'pluck' is faster on bigger one. Also, probably, it isn't much affected by the number of students so the recordings should be OK.